bridge/client/app/_components/ktb-evaluation-details/ktb-evaluation-details-utils.ts

```diff
@@ -102,3 +104,29 @@ export function createDataPoints(evaluationHistory: Trace[]): IDataPoint[] {
     .sort((a, b) => a.yElement.localeCompare(b.yElement));
   return [...scores, ...sortedResults];
 }
+
+export function parseSloFile(evaluationTraces: Trace[]): void {
+  for (const evaluationData of evaluationTraces) {
+    if (evaluationData?.data?.evaluation?.sloFileContent && !evaluationData.data.evaluation.sloFileContentParsed) {
+      try {
+        evaluationData.data.evaluation.sloFileContentParsed = parseYaml(
+          atob(evaluationData.data.evaluation.sloFileContent)
+        ) as SloConfig;
+        evaluationData.data.evaluation.score_pass =
+          evaluationData.data.evaluation.sloFileContentParsed.total_score?.pass?.split('%')[0] ?? '';
+        evaluationData.data.evaluation.score_warning =
+          evaluationData.data.evaluation.sloFileContentParsed.total_score?.warning?.split('%')[0] ?? '';
+        evaluationData.data.evaluation.compare_with =
+          evaluationData.data.evaluation.sloFileContentParsed.comparison.compare_with ?? '';
+        evaluationData.data.evaluation.include_result_with_score =
+          evaluationData.data.evaluation.sloFileContentParsed.comparison.include_result_with_score;
+        if (evaluationData.data.evaluation.comparedEvents) {
+          evaluationData.data.evaluation.number_of_comparison_results =
+            evaluationData.data.evaluation.comparedEvents?.length;
+        } else {
+          evaluationData.data.evaluation.number_of_comparison_results = 0;
+        }
+      } catch {}
+    }
+  }
+}
```

Comment:
I think can improve the function into two, like:
```
export function parseSloFile(evaluationTraces: Trace[]): void {
  const alreadyParsed = (trace: Trace): boolean => trace?.data?.evaluation?.sloFileContent
     && !trace.data.evaluation.sloFileContentParsed;
  const mapToEvaluation = (trace: Trace): Evaluation => {
    // Maybe checks for undefined
    return trace.data.evaluation;
  };
  evaluationTraces.filter(alreadyParsed).map(mapToEvaluation)
    .forEach(e => this.parseSloFileForEvaluation(e));
}

function parseSloFileForEvaluation(evaluation) {
  try {
    // parseYml
    // assigments
  } catch {  }
}
```

Reply:
For me this will not be an improvement. One for-loop is changed to three.

Reply:
If you look at it from a runtime perspective. Look at the code, it looks like this:
```
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx {
 xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx {
   xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx{
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx {
     xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    }
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
  }
 }
}
```
As a developer you need a long time to grasp the purpose of the function. In my opinion readability and maintainability goes before runtime, when there is no need for a fast execution. If we wanted a fast client we would write it in C, which also does too many CPU cycles compared to Assembler.

A good function should not have more than one indentation.